packages/aft/lib/src/commands/generate/generate_sdk_command.dart

```diff
@@ -134,7 +134,10 @@ class GenerateSdkCommand extends AmplifyCommand with GlobOptions {
       if (!await modelsDir.exists()) {
         exitError('Model directory ($modelsDir) does not exist');
       }
-      modelsDir = await _checkoutModelsRef(modelsDir, ref);
+      // Checkout models if modelsPath is a git repository.
+      if (File(p.join(modelsPath, '.git')).existsSync()) {
+        modelsDir = await _checkoutModelsRef(modelsDir, ref);
```

Comment:
Can you add a warn log, since this breaks expected behavior?

```dart
logger.warn('Unable to check out $ref since $modelsDir is not a git repo');
```